src/foam/nanos/approval/ApprovalDAO.java

```diff
@@ -0,0 +1,108 @@
+package foam.nanos.approval;
+
+import foam.core.FObject;
+import foam.core.X;
+import foam.dao.DAO;
+import foam.dao.ProxyDAO;
+import foam.mlang.sink.Sum;
+import foam.nanos.auth.User;
+import foam.nanos.ruler.Operations;
+import foam.nanos.approval.RoleApprovalRequest;
+import foam.nanos.approval.ApprovalStatus;
+import foam.nanos.approval.ApprovalRequest;
+
+import static foam.mlang.MLang.*;
+
+/**
+ * When a single request is approved, checks all other approved requests and calculates points.
+ * If points are sufficient ( >= requiredPoints ), removes all unused requests from approvalRequestDAO and
+ * Re-submits the object to its dao (daoKey), where a rule should be defined to determine further actions.
+ */
+public class ApprovalDAO
+  extends ProxyDAO {
+
+  public ApprovalDAO(X x,DAO delegate) {
+    setX(x);
+    setDelegate(delegate);
+  }
+
+  @Override
+  public FObject put_(X x, FObject obj) {
+    DAO requestDAO = ((DAO)x.get("approvalRequestDAO"));
+    ApprovalRequest old = (ApprovalRequest) requestDAO.find(obj);
+    ApprovalRequest request = (ApprovalRequest) getDelegate().put(obj);
+
+    if ( old != null && old.getStatus() != request.getStatus()
+      || old == null && request.getStatus() != ApprovalStatus.REQUESTED
+    ) {
+      DAO requests = ApprovalRequestUtil.getAllRequests(x, request.getObjId(), request.getClassification());
```

Comment:
In the current approval system, which is rule based, this logic is handled in support classes.

Reply:
See above about how this is all code from the approval system prior to Liquid.